src/atlas/array/helpers/ArrayForEach.h

```diff
@@ -114,28 +129,50 @@ constexpr auto argPadding() {
   }
 }
 
-template <typename... SlicerArgs, template <typename, int> typename View,
-          typename Value, int Rank, typename... ArrayViews>
+template<int Rank>
+struct GetRankImpl {
+  constexpr static int rank = Rank;
+};
+
+template<typename>
+struct GetRank;
+
+template<template <typename, int> typename View, typename Value, int Rank>
+struct GetRank<View<Value, Rank>> : GetRankImpl<Rank> {};
+template<template <typename, int> typename View, typename Value, int Rank>
+struct GetRank<View<Value, Rank>&> : GetRankImpl<Rank> {};
+template<template <typename, int> typename View, typename Value, int Rank>
+struct GetRank<const View<Value, Rank>&> : GetRankImpl<Rank> {};
+template<template <typename, int> typename View, typename Value, int Rank>
+struct GetRank<View<Value, Rank>&&> : GetRankImpl<Rank> {};
+template<template <typename, int> typename View, typename Value, int Rank>
+struct GetRank<const View<Value, Rank>&&> : GetRankImpl<Rank> {};
```

Comment:
I don't think the `const View` overloads are required. `View` itself is possible to be const